sql/2020/16_Ecommerce/pct_ampusage_bydevice_vendor.sql

```diff
@@ -2,30 +2,26 @@
 # 13_14: % of AMP enabled eCommerce Sites by device
 SELECT
   _TABLE_SUFFIX AS client,
-  vendor,
-  COUNTIF(app = 'AMP') AS AMPFreq,
-  SUM(COUNT(0)) OVER (PARTITION BY vendor) AS total,
-  ROUND(COUNTIF(app = 'AMP') * 100 / SUM(COUNT(0)) OVER (PARTITION BY vendor), 2) AS pct
-  FROM
-    `httparchive.technologies.2020_08_01_*`
-JOIN 
-(
-  SELECT 
-    _TABLE_SUFFIX AS client,
-    url, 
-    app as vendor
+  COUNT(DISTINCT url) AS freq,
+  total,
+  COUNT(DISTINCT url) / total AS pct
+FROM
+  `httparchive.technologies.2020_08_01_*`
+JOIN (
+  SELECT
+    _TABLE_SUFFIX,
+    COUNT(DISTINCT url) AS total
   FROM
     `httparchive.technologies.2020_08_01_*`
   WHERE
-    category = 'Ecommerce'
- )
-USING 
-  (url)
+     category = 'Ecommerce'
+  GROUP BY
+    _TABLE_SUFFIX
+    )
+USING
+  (_TABLE_SUFFIX)
+WHERE
+  app = 'AMP'
 GROUP BY
-  client, 
-  vendor
-ORDER BY
-  total desc,
-  AMPFreq desc,
-  Vendor
-  
\ No newline at end of file
+  client,
+  total
```

Comment:
This query is not correct. It is getting the number of AMP sites (not necessarily eCommerce) and then using that as a percentage of eCommerce.

So if you 20 AMP sites, non of which are on Ecommerce sites, and you have 100 Ecommerce sites in total, then this query will report that 20% of Ecommerce sites are on AMP.

I would suggest the following is what you are looking for instead, where I join to ensure only AMP Ecommerce sites are chosen eligible:

```sql
#standardSQL
# 13_14: % of AMP enabled eCommerce Sites by device
SELECT
  _TABLE_SUFFIX AS client,
  COUNT(DISTINCT url) AS freq,
  total,
  COUNT(DISTINCT url) / total AS pct
FROM
  `httparchive.technologies.2020_08_01_*`
JOIN (
  SELECT
    _TABLE_SUFFIX,
    url
  FROM
    `httparchive.technologies.2020_08_01_*`
  WHERE
     category = 'Ecommerce'
    )
USING
  (_TABLE_SUFFIX, url)
JOIN (
  SELECT
    _TABLE_SUFFIX,
    COUNT(DISTINCT url) AS total
  FROM
    `httparchive.technologies.2020_08_01_*`
  WHERE
     category = 'Ecommerce'
  GROUP BY
    _TABLE_SUFFIX
    )
USING
  (_TABLE_SUFFIX)
WHERE
  app = 'AMP'
GROUP BY
  client,
  total
ORDER BY
  client
```